buildkitd/entrypoint.sh

```diff
@@ -252,6 +252,13 @@ echo "Detected container architecture is $(uname -m)"
 "$@" &
 execpid=$!
 
+stop_buildkit() {
+  echo "SIGTERM received. Shutting down buildkit..."
+  kill -SIGTERM "$execpid"
+}
+
+trap stop_buildkit SIGTERM
```

Comment:
Maybe also add `SIGQUIT` and `SIGINT` for completion's sake?

```suggestion
trap stop_buildkit SIGTERM SIGQUIT SIGINT
```